pkg/ledger/ledger_monitor.go

```diff
@@ -123,6 +132,33 @@ func (m *Monitor) Start(ctx context.Context) {
 	})
 }
 
+func (m *Monitor) isHealthy() bool {
+	return m.healthy.IsSet()
+}
+
+func (m *Monitor) listenSocket(sock string) error {
```

Comment:
Would also recommend that we use a mux here and listen for `/health` for the healthcheck, to make this more extensible down the road.

Reply:
agreed.